src/math/MultiplierPointMath.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT-1.0
+pragma solidity ^0.8.26;
+
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { IStakeConstants } from "../interfaces/IStakeConstants.sol";
+
+/**
+ * @title MultiplierPointMath
+ * @author Ricardo Guilherme Schmidt <ricardo3@status.im>
+ * @notice Provides mathematical operations and utilities for managing multiplier points in the staking system.
+ */
+abstract contract MultiplierPointMath is IStakeConstants {
+    /// @notice One (mean) tropical year, in seconds.
+    uint256 public constant YEAR = 365 days;
+    /// @notice Accrued multiplier points maximum multiplier.
+    uint256 public constant MAX_MULTIPLIER = 4;
+    /// @notice Multiplier points annual percentage yield.
+    uint256 public constant MP_APY = 100;
+    /// @notice Multiplier points accrued maximum percentage yield.
+    uint256 public constant MP_MPY = MAX_MULTIPLIER * MP_APY;
+    /// @notice Multiplier points absolute maximum percentage yield.
+    uint256 public constant MP_MPY_ABSOLUTE = 100 + (2 * (MAX_MULTIPLIER * MP_APY));
+    /// @notice The accrue rate period of time over which multiplier points are calculated.
+    uint256 public constant ACCRUE_RATE = 12 seconds;
+    /// @notice Minimal value to generate 1 multiplier point in the accrue rate period (rounded up).
+    uint256 public constant MIN_BALANCE = (((YEAR * 100) - 1) / (MP_APY * ACCRUE_RATE)) + 1;
```

Comment:
Where does this `100` come from? i It's not the `MP_APY`
Can we have this live in a constant as well?

Reply:
Okay I see, this makes more sense now.

I still think that the `MIN_BALANCE` is something we should avoid.
We should rather check if MPs greater > 0 will be generated with the given stake amount and if not, simply not update the lastUpdateTime, so next time we calculate MPs we still calculate from the original last update.

See the comment here: https://github.com/vacp2p/staking-reward-streamer/pull/109#discussion_r1917277757